contracts/contracts/token/OUSD.sol

```diff
@@ -584,20 +609,111 @@ contract OUSD is Initializable, InitializableERC20Detailed, Governable {
             ? MAX_SUPPLY
             : _newTotalSupply;
 
-        _rebasingCreditsPerToken = _rebasingCredits.divPrecisely(
-            _totalSupply.sub(nonRebasingSupply)
-        );
+        _rebasingCreditsPerToken =
+            (_rebasingCredits * 1e18) /
+            (_totalSupply - nonRebasingSupply);
 
         require(_rebasingCreditsPerToken > 0, "Invalid change in supply");
 
-        _totalSupply = _rebasingCredits
-            .divPrecisely(_rebasingCreditsPerToken)
-            .add(nonRebasingSupply);
+        _totalSupply =
+            ((_rebasingCredits * 1e18) / _rebasingCreditsPerToken) +
+            nonRebasingSupply;
 
         emit TotalSupplyUpdatedHighres(
             _totalSupply,
             _rebasingCredits,
             _rebasingCreditsPerToken
         );
     }
+
+    function delegateYield(address from, address to)
+        external
+        onlyGovernor
+    {
+        require(from != to, "Cannot delegate to self");
```

Comment:
🟥 Lol, we need checks that from and to are not zero. Otherwise we could think we have no yield delegation setup and multi delegate.

Reply:
Hah very nice catch! fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/5f63650cfe5d629cd35445d097fdf4cda4d25017

Reply:
really fixed here: https://github.com/OriginProtocol/origin-dollar/pull/2298/commits/4e327faa095e83c89a9c61f48921c1c751e1d457